src/Field/Configurator/ChoiceConfigurator.php

```diff
@@ -107,6 +107,11 @@ public function configure(FieldDto $field, EntityDto $entityDto, AdminContext $c
         // Translatable choice don't need to get flipped
         $flippedChoices = $areChoicesTranslatable ? $choices : array_flip($this->flatten($choices));
         foreach ((array) $fieldValue as $selectedValue) {
+            if ($selectedValue instanceof \BackedEnum) {
+                $selectedValue = $selectedValue->value;
+            } elseif ($selectedValue instanceof \UnitEnum) {
+                $selectedValue = $selectedValue->name;
+            }
```

Comment:
Probably preferable to have something like this, to have easier condition clauses with less complexity compared to if/elseif statements. [Avoid them when possible](https://softwareengineering.stackexchange.com/questions/206816/clarification-of-avoid-if-else-advice). 
```suggestion
            $selectedValue = match(true) {
                $selectedValue instanceof \BackedEnum => $selectedValue->value,
                $selectedValue instanceof \UnitEnum => $selectedValue->name,
                default => $selectedValue
            };
```